File: frontend/src/pages/app/api/projectData/categories/BibCategory.ts

export interface BibCategory {
  Name:                  string
  CitaviType:            string
  CitaviNecessaryFields: string[] //z.B. nur dieser Typ wenn doi existiert
  Fields:                Field[]
  CiteFields:            Field[]
  Example: string
}

export interface Field {
  Field:           string
  Italic:            boolean
  Prefix:           string
  Suffix:           string
  TexValue:         boolean
  CitaviAttributes: string[]
}

File: services/app/internal/database/bib_category.go
package database

type BibCategory struct {
	Name                  string
	CitaviType            string
	CitaviNecessaryFields []string //z.B. nur dieser Typ wenn doi existiert
	Fields                []Field
	CiteFields            []Field
}

type Field struct {
	Field            string
	Italic           bool
	Prefix           string
	Suffix           string
	TexValue         bool
	CitaviAttributes []string
}
